official/recommendation/data_preprocessing.py

```diff
@@ -644,10 +661,43 @@ def input_fn(params):
   return input_fn, record_dir, batch_count
 
 
+_SYNTHETIC_BATCHES_PER_EPOCH = 2000
+
+
+def make_train_synthetic_input_fn():
+  """Construct training input_fn that uses synthetic data."""
+  def input_fn(params):
+    """Generated input_fn for the given epoch."""
+    batch_size = params["batch_size"]
+    num_users = params["num_users"]
+    num_items = params["num_items"]
+
+    users = tf.random_uniform([batch_size], dtype=tf.int32, minval=0,
+                              maxval=num_users)
+    items = tf.random_uniform([batch_size], dtype=tf.int32, minval=0,
+                              maxval=num_items)
+    labels = tf.random_uniform([batch_size], dtype=tf.int32, minval=0,
+                               maxval=2)
+
+    data = {
+        movielens.USER_COLUMN: users,
+        movielens.ITEM_COLUMN: items,
+    }, labels
+    dataset = tf.data.Dataset.from_tensors(data).repeat(
```

Comment:
Didn't you want to have some interleaving here? So as to better mimic the threading behavior?

Reply:
I have no map_func, so there is nothing to interleave. Also, with synthetic data, I'm not really trying to measure the performance of the input pipeline.